admin/test/test_merge_pr.py

```diff
@@ -0,0 +1,212 @@
+    booleans,
+    dictionaries,
+    fixed_dictionaries,
+    integers,
+    just,
+    lists,
+    one_of,
+    sampled_from,
+    text,
+    )
+from pyrsistent import pmap
+
+from flocker.testtools import TestCase
+
+from admin import merge_pr
+
+
+SCRIPT_FILENAME = 'merge-pr'
+SCRIPT_FILE = os.path.join(os.path.dirname(os.path.dirname(__file__)),
+                           SCRIPT_FILENAME)
+
+
+def run_script(args):
+    return subprocess.check_output([SCRIPT_FILE] + args)
+
+
+class SmokeTests(TestCase):
+    """
+    Basic tests of running the script.
+    """
+
+    def test_help(self):
+        output = run_script(['--help'])
+        self.assertIn("Merge a branch when all the tests pass.", output)
+
+
+class URLTests(TestCase):
+    """
+    Test for URL manipulation.
+    """
+
+    def test_url_path(self):
+        """
+        Correct path from a full URL.
+        """
+        path = '/ClusterHQ/flocker/pull/1717'
+        self.assertEqual(path, merge_pr.url_path('https://github.com' + path))
+
+    def test_url_path_no_hostname(self):
+        """
+        Correct path from a URL path.
+        """
+        path = '/ClusterHQ/flocker/pull/1717'
+        self.assertEqual(path, merge_pr.url_path(path))
+
+    def test_url_path_parts(self):
+        """
+        Correct segments from a full URL.
+        """
+        path_parts = ['ClusterHQ', 'flocker', 'pull', '1717']
+        self.assertEqual(
+            [''] + path_parts,
+            merge_pr.url_path_parts(
+                'https://github.com/' + '/'.join(path_parts)))
+
+    def test_url_path_parts_no_hostname(self):
+        """
+        Correct segments from a URL path.
+        """
+        path_parts = ['ClusterHQ', 'flocker', 'pull', '1717']
+        self.assertEqual(
+            [''] + path_parts,
+            merge_pr.url_path_parts('/' + '/'.join(path_parts)))
+
+    def test_pr_api_url(self):
+        """
+        Correct API URL for a full URL.
+        """
+        self.assertEqual(
+            'https://api.github.com/repos/ClusterHQ/flocker/pulls/1717',
+            merge_pr.pr_api_url_from_web_url(
+                'https://github.com/ClusterHQ/flocker/pull/1717'))
+
+
+# The max here corresponds to the latest date that datetime supports
+datetimes = integers(max_value=253402300799).map(datetime.fromtimestamp)
+"""Strategy for generating `datetime` objects."""
+
+
+commit_statuses = fixed_dictionaries(
+    {'updated_at': datetimes,
+     'state': text(),
+     'context': text(),
+     'target_url': text(),
+     })
+"""Strategy for generating GitHub commit status dicts."""
+
+
+jenkins_results = sampled_from(merge_pr.JenkinsResults.iterconstants())
+"""Strategy for generating JenkinsResults values."""
+
+
+class StatusesTests(TestCase):
+    """
+    Tests for interpretation of commit statuses.
+
+    https://developer.github.com/v3/repos/statuses/
+    """
+
+    @given(commit_statuses)
+    def test_final_status_one(self, status):
```

Comment:
:question: nit: docstrings. These tests are pretty simple and straightforward, but flocker codebase is pretty uniform in having docstrings on all test methods:

```
 ♦ ~/dev/flocker (master)$ git grep -A 1 'def test_' | grep '"""' | wc
    2049    4308  103346
 ♦ ~/dev/flocker (master)$ git grep -A 1 'def test_' | grep 'def test_' | wc
    2085    6549  166892
```

So I guess only ~36/2085 tests are missing docstrings. It's not technically in our handbook, so ultimately your call if it's worth adding them.